### DISCUSSION TITLE
Exclude fields from scanner if their name match a regex

### DISCUSSION THREAD

**Discussion by phuonghuynh** [mapstruct/mapstruct#3650] [Q&A] [answered]
Hi,

How to exclude all fields from both source and target that have names match a regex, or annotated by a @CustomAnnotation ?

**Comment by phuonghuynh** [comment, marked as answer]
For those who looking for a simple solution, by using MapStruct SPI, I hope MapStruct will implement this feature into next releases.

```
public class ExcludeFieldEndsWith_ extends DefaultAccessorNamingStrategy {

    @Override
    public boolean isGetterMethod(ExecutableElement method) {
        var name = this.getPropertyName(method);
        if ("ExcludeFieldEndsWith_".equalsIgnoreCase(name)) {
            return false;
        }
        return super.isGetterMethod(method);
    }

    @Override
    public boolean isSetterMethod(ExecutableElement method) {
        var name = this.getPropertyName(method);
        if ("ExcludeFieldEndsWith_".equalsIgnoreCase(name)) {
            return false;
        }
        return super.isSetterMethod(method);
    }

    @Override
    public String getPropertyName(ExecutableElement getterOrSetterMethod) {
        var name = super.getPropertyName(getterOrSetterMethod);
        return name.endsWith("_") ? "ExcludeFieldEndsWith_" : name;
    }
}

```

**Comment by thunderhook** [comment]
Using a regex has already been asked and is not planned to be implemented. The way to go is to implement a `NamingAccessorStrategy`, as you ended up doing because it is much more flexible than a regex, see also https://github.com/mapstruct/mapstruct/issues/292#issuecomment-1826721068

About using a custom annotation: I think you mean an annotation that you put on your models and get ignored like `@JsonIgnore`? This was discussed and decided against in #1152 in favour of the `NamingAccessorStrategy`. I don't know the exact reason, but one I could think of is that you can only use this feature if you have control over the classes (and they are not from external projects), and to reduce the clutter inside the models.

So I don't think that MapStruct will implement something like this in the near future.

What might come is some kind of configuration file about different getter/wither styles. That's not exactly a solution to _your_ problem, but it might solve some custom naming accessor strategies, for more details see: https://github.com/mapstruct/mapstruct/issues/3211#issuecomment-1529044934
